### ISSUE TITLE
 The 'main' function is in error: Invalid script file name configuration. The 'scriptFile' property is set to a file that does not exist.

### ISSUE THREAD

**Issue by Notaduck** [lukas-reining/nestjs-azure-func-trigger#1]
Hi There. I hope it is okay to post this as an issue, however this is probably my own doing.

I have a nestjs project which I'm trying to deploy as an azure function time trigger.

What I've done so far is to install the `@nestjs/azure-func-http` and `@nestjs/azure-func-trigger`.

1. `./main/`
  
`function.json`:
```json
{
  "bindings": [
    {
      "name": "trigger",
      "type": "timerTrigger",
      "direction": "in",
      "schedule": "*/10 * * * * *"
    }
  ],
  "scriptFile": "../dist/FunctionName/index.js"
}

```

`index.ts`:
```ts
import { Context } from '@azure/functions';
import { AzureFunctionTriggerAdapter } from 'nestjs-azure-func-trigger';
import { createApp } from '../src/main.azure';

export default async function (context: Context): Promise<void> {
  return AzureFunctionTriggerAdapter.handle(createApp, context);
}
```

3.`./src`
`main.azure.ts`

```ts
import { INestApplication, Logger } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';

export async function createApp(): Promise<INestApplication> {
  const app = await NestFactory.create(AppModule);
  app.setGlobalPrefix('api');
  
  await app.init();
  return app;
}

```

4. `./src/contacts`

`contacts.service.ts`
```ts

import { AzureFunctionContext } from 'nestjs-azure-func-trigger/dist/decorators/azure-function-context';
import { AzureFunctionTrigger } from 'nestjs-azure-func-trigger/dist/decorators/azure-function-trigger';
import { Context } from '@azure/functions';

@Injectable()
export class ContactsService {
  .........
  .........
  @AzureFunctionTrigger('FunctionName')
  public async importDatabaseContactToDynamics() {
  ..........
  .........
  }
```

------------------------------------------------------ 

I have tried to take a look at the `dist` folder but `../dist/FunctionName/index.js` is not there and I'm not exactly sure what is going on.

![Screenshot 2023-08-07 at 15 08 07](https://github.com/lukas-reining/nestjs-azure-func-trigger/assets/9454064/ce147d58-6462-494a-b807-c0d58458a7bf)

So, the time trigger called `FunctionName` doesn't seem to be in `dist` at all and `importDatabaseContactToDynamics` is never executed.

**Comment by Notaduck** [comment]
I just needed to create another /FunctionName/ folder with the appropriate files, this was not an issue in the end.
However the `context` is `undefined` and the `test` holds the context for some reason.

```js
  @AzureFunctionTrigger('data-glider')
  public timerWithContext(
    test: string,
    @AzureFunctionContext() context: Context,
  ) {
    this.logger.log('executed');
    context.log('TimerWithContext ------------');
  }
```

**Comment by lukas-reining** [comment]
Hey @Notaduck,

sorry for answering so late, I did not see this issue as I got this repo a bit out of my eyes.
Great that you have been able to resolve the issue with getting it to work yourself.

For the second issue you have mentioned, I have released a fix now.
I accidentally always provided the context as first argument if the decorator was used, regardless of the argument it was put on.
Actually the example (which I changed now) did not make sense too, the test string could never be populated anyways.
Anyways I resolved this issue now, and the context is provided to all the arguments annotated with `@AzureFunctionContext()`.

In the following example the context will be filled like: `undefined, context, context, undefined, context`

```ts
  @AzureFunctionTrigger('data-glider')
  public timerWithContext(
    test1: string,
    @AzureFunctionContext() context1: Context,
    @AzureFunctionContext() context2: Context,
    test2: string,
    @AzureFunctionContext() context3: Context,
  ) {
    this.logger.log('executed');
    context.log('TimerWithContext ------------');
  }
```

Also Azure Functions SDK V4 can also be used now.

Are you still using this lib? If so does it work as expected now @Notaduck?
